src/delegation/store/memory.rs

```diff
@@ -69,36 +73,77 @@ use web_time::SystemTime;
 /// linkStyle 6 stroke:orange;
 /// linkStyle 1 stroke:orange;
 /// ```
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone)]
 pub struct MemoryStore<
     DID: did::Did + Ord = did::preset::Verifier,
     V: varsig::Header<C> = varsig::header::Preset,
     C: Codec + TryFrom<u64> + Into<u64> = varsig::encoding::Preset,
 > {
-    ucans: BTreeMap<Cid, Delegation<DID, V, C>>,
+    inner: Arc<RwLock<MemoryStoreInner<DID, V, C>>>,
```

Comment:
A couple questions, oh wise Rustacean 🦀

## Generality

I guess `RwLock` here because it's the most general option? Is this a typical interface for libraries (as opposed to applications)? It really makes me want HKTs 😛

## Deadlocks

From the `std` docs:

> The priority policy of the lock is dependent on the underlying operating system’s implementation, and this type does not guarantee that any particular policy will be used.
> [...]
> Potential deadlock example

That's... unsettling. Are there ways to guard against starvation other than "don't use a bad OS"?

Reply:
> oh wise Rustacean 🦀

:joy: :man_facepalming: :sweat_smile: 

> Deadlocks [...] That's... unsettling. Are there ways to guard against starvation other than "don't use a bad OS"?

It is unsettling! And you're wise in reading the documentation more than I have :stuck_out_tongue: 
I chose RwLock because it should be faster than Mutex or equally fast in all cases, but admittedly, the deadlock potential on some OSes worries me, too. And performance shouldn't matter *too* much here, and if someone is worried about that, they could/should write a store implementation based on `DashMap`.
Should we change it to Mutex?

Reply:
Reading up a bit I think I had some misconceptions about RwLocks. They can be slower than Mutexes in some cases. Some people recommend "Use Mutex unless you know what you're doing".
And I clearly don't! So let's switch to mutex :+1:

Reply:
By the way -
> Is this a typical interface for libraries (as opposed to applications)?

I'm not sure I understand. The RwLock/Mutex is totally an implementation detail and shouldn't leak to the interface. It's a private field.
The `MemoryStore`/`MemoryStoreInner` is a typical way of structuring code.